ktor-client/ktor-client-tests/build.gradle.kts

```diff
@@ -166,10 +131,53 @@ if (!ideaActive) {
 }
 
 rootProject.allprojects {
-    if (path.contains("ktor-client")) {
-        val tasks = tasks.matching { it.name in testTasks }
-        configure(tasks) {
-            dependsOn(startTestServer)
+    if (!path.contains("ktor-client")) {
+        return@allprojects
+    }
+    val tasks = tasks.matching { it.name in testTasks }
+    configure(tasks) {
+        dependsOn(startTestServer)
+        kotlin.sourceSets {
+            if (!(rootProject.ext.get("native_targets_enabled") as Boolean)) {
+                return@sourceSets
+            }
+            if (ideaActive) {
+                if (name == "posixTest") {
+                    getByName(name) {
+                        dependencies {
+                            val hostname: String by project.ext
+                            api(project(":ktor-client:ktor-client-curl"))
+
+                            if (!hostname.startsWith("win")) {
+                                api(project(":ktor-client:ktor-client-cio"))
+                            }
+                        }
+                    }
+                }
+                return@sourceSets
+            }
+            if (name in listOf("macosX64Test", "linuxX64Test")) {
```

Comment:
`when` could be used

Reply:
yes, but it adds one more level of nesting